backend/inspirehep/disambiguation/tasks.py

```diff
@@ -70,26 +75,127 @@ def disambiguate_signatures(self, clusters):
     db.session.commit()
 
 
+def match_literature_author_with_config(author_data, matcher_config):
+    matched_records = [
+        matched_record
+        for matched_record in match(author_data, matcher_config)
+        if matched_record.get("inner_hits")
+    ]
+    matched_authors_refs = match_literature_author_ids(matched_records)
+    return matched_authors_refs, matched_records
+
+
+def match_literature_author_with_validator(validators, author_data, matched_references):
+    for validator in validators:
+        valid_matches = (
+            match for match in matched_references if validator(author_data, match)
+        )
+        matched_valid_authors_references = match_literature_author_ids(valid_matches)
+        if matched_valid_authors_references:
+            return matched_valid_authors_references
+
+
+def match_literature_author_ids(matched_records):
+    matched_authors_refs = set()
+    for record_authors in matched_records:
+        matched_authors_refs.update(
+            get_value(
+                record_authors, "inner_hits.authors.hits.hits._source.record.$ref", None
+            )
+        )
+    if matched_authors_refs and len(matched_authors_refs) == 1:
+        return matched_authors_refs
+
+
+def assign_reference_to_author_if_unambiguous_match(matched_authors_refs, author):
```

Comment:
this is a misnomer now, I think it would be better to leave the length check here (which would work together with my other suggestions).